File: js/src/0.0-Basics/0.3-build-up-logical-thinking/0.3.21-pattern.js
function log(val) {
    process.stdout.write(String(val));
}

function pattern_21() {
    let column = 5;
    let row = 5;

    for (let y = 1; y < row + 1; y++) {
        let counter = 1;

        for (let x = column; x > 0; x--) {
            if (x <= y) {
                log(counter);

                counter = counter + 1;
            } else {
                log(' ');
            }
        }
        counter = 1;
        console.log();
    }
}

console.clear();
pattern_21();

File: python/00-Basics/00.03-build-up-logical-thinking/0.3.21-pattern.py
from utils.clear import clear

def pattern_21():
    row = 5
    column = 5

    counter = 1

    for i in range(row, 0, -1):
        for j in range(1, column + 1):
            if j >= i:
                print(counter, end="")
                counter = counter + 1
            elif j < i:
                print(" ", end="")

        counter = 1
        print()

clear()
pattern_21()
